### PR TITLE
Update aurora-workspace dependency

### PR DESCRIPTION
**Pull request by karim-en**
## Description
Update `aurora-workspace` dependency.
There was a conflict with `borsh` dependency version, so the `near-sdk  4.1.1` was replaced with the forked one [v4.1.1-fix-deps](https://github.com/aurora-is-near/near-sdk-rs/releases/tag/v4.1.1-fix-deps).
Also, the `aurora-engine-migration-tool` `rev` point to commit that wasn't yet merged into the master.

## Performance / NEAR gas cost considerations
For unknown reasons, the test `test_migration_state` had failed after this update, so it was fixed by increasing the gas burnt check from `1984` to `1987`

### CODE REVIEW

**Review by olga24912 — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -7,14 +7,15 @@ repository = "https://github.com/aurora-is-near/aurora-eth-connector"
 license = "CC0 1.0"
 
 [workspace.dependencies]
-aurora-engine-types = { git = "https://github.com/aurora-is-near/aurora-engine.git", branch = "develop", default-features = false }
+aurora-engine-types = { git = "https://github.com/aurora-is-near/aurora-engine.git", tag = "3.3.1", default-features = false }
```
Comment (mrLSD) [inline Cargo.toml:10]: Why not `v3.5.0`?

**Review by mrLSD — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (olga24912) [reply Cargo.toml:10]: To be consistent with aurora-workspace: https://github.com/aurora-is-near/aurora-workspace/blob/main/Cargo.toml#L24

Comment (mrLSD) [reply Cargo.toml:10]: Ok then, released:
https://github.com/aurora-is-near/aurora-workspace/releases/tag/0.4.1

Please update `aurora-engine` version and `aurora-workspace`.

Comment (karim-en) [reply Cargo.toml:10]: Could you please also release the migration tools with the deps fixes? So we can relplase the rev commit with tag

Comment (mrLSD) [reply Cargo.toml:10]: Yes, sure.

Comment (mrLSD) [reply Cargo.toml:10]: https://github.com/aurora-is-near/aurora-engine-migration-tool/pull/20

Comment (karim-en) [reply Cargo.toml]: 721ac9201beb6fe3baadec1066b5882908c1dc4e

**Pull request merged**